bmap/src/discarder.rs

```diff
@@ -35,6 +42,44 @@ impl<R: Read> SeekForward for Discarder<R> {
         Ok(())
     }
 }
+//Async implementation
+
+pub struct AsyncDiscarder<R: AsyncRead> {
+    reader: R,
+}
+
+impl<R: AsyncRead + Debug> AsyncDiscarder<R> {
+    pub fn new(reader: R) -> Self {
+        Self { reader }
+    }
+
+    pub fn into_inner(self) -> R {
+        self.reader
+    }
+}
+
+impl<R: AsyncRead + Unpin> AsyncRead for AsyncDiscarder<R> {
+    fn poll_read(
+        mut self: Pin<&mut Self>,
+        cx: &mut Context<'_>,
+        buf: &mut tokio::io::ReadBuf<'_>,
+    ) -> Poll<Result<(), std::io::Error>> {
+        Pin::new(&mut self.reader).poll_read(cx, buf)
+    }
+}
+#[async_trait(?Send)]
+impl<R: AsyncRead + Unpin> AsyncSeekForward for AsyncDiscarder<R> {
+    async fn async_seek_forward(&mut self, forward: u64) -> IOResult<()> {
+        let mut buf = [0; 4096];
+        let mut left = forward as usize;
+        while left > 0 {
+            let toread = left.min(buf.len());
+            let r = block_on(self.read(&mut buf[0..toread]))?;
```

Comment:
don't call block_on in an async method; as that starts a new executors that blocks the overall executor; Instead do a proper async read

Reply:
right so that explains why in my testing CPU usage would peak quite high